rosters.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+
+import argparse
+import csv
+import os
+import re
+import requests
+from getpass import getpass
+
+# ANSI color codes
+GREEN = "\033[01;92m"
+RED = "\033[01;91m"
+END = "\033[0m"
+
+def download_roster(course_id, canvas_api_key):
+    response = requests.get(f'https://courseworks2.columbia.edu/api/v1/courses/{course_id}/students',
+            headers={'Authorization': f'Bearer {canvas_api_key}'}).json()
+
+    response = list(filter(lambda x: x['sortable_name'] != 'Student, Test', response))
+
+    return [(user['login_id'], user['sortable_name']) for user in response]
+
+# Turns ('abc', 'def', '') into ('abc','def')
+def discard_last_if_empty(elements):
+    if len(elements) == 0 or len(elements[-1]) > 0:
+        return elements
+    else:
+        return elements[:-1]
+
+
```

Comment:
Would be good to have documentation about what format this expects the roster to be in and what format it returns it in -- those have probably been the main source of issues with this repo over the past couple years